src/Data/List/Relation/OrderPreservingEmbedding.agda

```diff
@@ -0,0 +1,127 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Inductive Definition of Order Preserving Embeddings
+------------------------------------------------------------------------
+
+module Data.List.Relation.OrderPreservingEmbedding where
+
+open import Data.Empty
+open import Data.Nat.Base
+open import Data.Nat.Properties
+open ≤-Reasoning
+open import Data.List.Base
+open import Function
+import Function.Injection as F
+open import Relation.Binary
+open import Relation.Binary.PropositionalEquality
+
+
+------------------------------------------------------------------------
+-- Type and Basic combinators
+
+module _ {a} {A : Set a} where
+
+  infix 3 _↣_
+  data _↣_ : Rel (List A) a where
+    base : [] ↣ []
+    weak : ∀ {xs y ys} → xs ↣ ys → xs ↣ (y ∷ ys)
+    copy : ∀ {x xs ys} → xs ↣ ys → (x ∷ xs) ↣ (x ∷ ys)
+
+  infix 4 []↣_
+  []↣_ : ∀ xs → [] ↣ xs
+  []↣ []     = base
+  []↣ x ∷ xs = weak ([]↣ xs)
+
+------------------------------------------------------------------------
+-- Interaction with List operations
+
+  length-↣-≤ : ∀ {xs ys} → xs ↣ ys → length xs ≤ length ys
+  length-↣-≤ base     = ≤-refl
+  length-↣-≤ (weak p) = ≤-step (length-↣-≤ p)
+  length-↣-≤ (copy p) = s≤s (length-↣-≤ p)
+
+  ++-↣ : ∀ {xs ys us vs} → xs ↣ ys → us ↣ vs → xs ++ us ↣ ys ++ vs
```

Comment:
This and other below, should all use the standard superscript +/- notation for introduction/elimination forms. Also if they are going to have the `↣` symbol in their name, it should be at the front of the name to match the membership modules (ditto with the length proof above).